# Environmental bioinformatics final project

Team: Max Jahns, Kate Lane, Katie Halloran

Paper: Shibl, A.A., et al. Diatom modulation of select bacteria through use of two unique secondary metabolites. Proc. Nat. Acad. Sci. 117, 27445-27455 (2020). https://doi.org/10.1073/pnas.2012088117

For this project, we proposed reproducing the analysis performed by Shibl and colleagues of data pulled from 

In this paper, Shibl and colleagues examined how phytoplankton modulate and control their bacterial community using excreted metabolites in response to community interactions. They use metagenomics, (meta)transcriptomics, and metabolomics to investigate interactions between the marine diatom Asterionellopsis glacialis and its bacterial consortium. To reproduce their work, we will assemble bacterial genomes from metagenomes, analyze differential gene expression during coculture using the metatranscriptomes, and complement the omics results with metabolite mass spectrometry data. Finally, we will replicate their analysis of the distribution of an azelaic acid transcriptional regulator with a hidden Markov model against the Tara Oceans bacterial metagenome dataset. More specifically, for our project, we plan to replicate the data analysis and visualization underlying Figures 1, 2, and 5d. Figure 3 illustrates biochemical processes, Figure 4 analyzes microscopy data which is not bioinformatically relevant, and Figure 5a-c uses experimental data. All genomic data is available on NCBI, and mass spectral datasets are on MassIVE database. Genomic data in this study  totals about 250Gb, spectral data is 9Kb. We would like to discuss with our faculty mentor if it would be best to subset data to use in our analysis.
